httpcache_test.go

```diff
@@ -1355,6 +1392,54 @@ func TestStaleIfErrorResponseLifetime(t *testing.T) {
 	}
 }
 
+func TestStaleIfErrorKeepsStatus(t *testing.T) {
```

Comment:
What is the difference in what this test is testing compared to `TestCachedErrorsKeepStatus`?

Reply:
The other is for the case when a non-200 status is returned because it had etag/last-modified and it was validated. This one is for the case where we use a stale cached response because the request errors and the request had stale-if-error set.

Having written that, I see the test names (at least TestCachedErrorsKeepStatus) aren't so clear ... if you have suggestions they're welcome, or I'll think some more about it :)

Reply:
You could keep names as is, but add a comment documenting what kind of behavior is being tested. I admit, it's quite hard to deduct from looking at the test code what it intends to test.